client/src/client/client_connection.rs

```diff
@@ -31,7 +31,5 @@ pub trait ClientConnection: Sized {
     /// Return the inner Futures items
     ///
     /// Consumes the connection and allows for future based operations afterward.
-    fn unwrap(
-        self,
-    ) -> (Core, Box<Future<Item = Self::MessageStream, Error = io::Error>>, Box<DnsStreamHandle<Error = ClientError>>);
+    fn new_stream(&self, handle: &Handle) -> ClientResult<(Box<Future<Item = Self::MessageStream, Error = io::Error>>, Box<DnsStreamHandle<Error = ClientError>>)>;
```

Comment:
There's a positive and negative to this new construct. TCP streams will always be reconnected if they fail, but obviously they will have to go through the 3way handshake again...